R/oc_forward.R

```diff
@@ -0,0 +1,328 @@
+    } else if (is.call(min_confidence_)) {
+      min_confidence <- eval(min_confidence_)
+    }
+    if (!is.null(min_confidence)) min_confidence <- as.list(min_confidence)
+
+    no_annotations_ <- eval(substitute(alist(no_annotations)))[[1]]
+    if (is.symbol(no_annotations_)) {
+      no_annotations_ <- data[[deparse(no_annotations_)]]
+      if (!is.null(no_annotations_)) no_annotations <- no_annotations_
+    } else if (is.call(no_annotations_)) {
+      no_annotations <- eval(no_annotations_)
+    }
+    if (!is.null(no_annotations)) no_annotations <- as.list(no_annotations)
+
+    no_dedupe_ <- eval(substitute(alist(no_dedupe)))[[1]]
+    if (is.symbol(no_dedupe_)) {
+      no_dedupe_ <- data[[deparse(no_dedupe_)]]
+      if (!is.null(no_dedupe_)) no_dedupe <- no_dedupe_
+    } else if (is.call(no_dedupe_)) {
+      no_dedupe <- eval(no_dedupe_)
+    }
+    if (!is.null(no_dedupe)) no_dedupe <- as.list(no_dedupe)
+
+    abbrv_ <- eval(substitute(alist(abbrv)))[[1]]
+    if (is.symbol(abbrv_)) {
+      abbrv_ <- data[[deparse(abbrv_)]]
+      if (!is.null(abbrv_)) abbrv <- abbrv_
+    } else if (is.call(abbrv_)) {
+      abbrv <- eval(abbrv_)
+    }
+    if (!is.null(abbrv)) abbrv <- as.list(abbrv)
+
+    output <- match.arg(output)
+
+    # Ensure that query column always exists
+    add_request <- TRUE
+
+    if (output == "short") {
+      no_annotations <- TRUE
+    }
+
+    if (bind_cols == FALSE) {
+      results_list <- oc_forward(
+        placename = placename,
+        key = key,
+        return = "df_list",
+        bounds = bounds,
+        countrycode = countrycode,
+        language = language,
+        limit = limit,
+        min_confidence = min_confidence,
+        no_annotations = no_annotations,
+        no_dedupe = no_dedupe,
+        no_record = no_record,
+        abbrv = abbrv,
+        add_request = add_request
+      )
+      results <- dplyr::bind_rows(results_list)
+      if (output == "short") {
+        results <-
+          dplyr::select(results, query, lat, lng, formatted)
+      } else {
+        results <-
+          dplyr::select(results, query, lat, lng, dplyr::everything())
+      }
+    } else {
+      results_nest <-
+        dplyr::mutate(
+          data,
+          op =
+            oc_forward(
+              placename = placename,
+              key = key,
+              return = "df_list",
+              bounds = bounds,
+              countrycode = countrycode,
+              language = language,
+              limit = limit,
+              min_confidence = min_confidence,
+              no_annotations = no_annotations,
+              no_dedupe = no_dedupe,
+              no_record = no_record,
+              abbrv = abbrv,
+              add_request = add_request
+            )
+        )
+
+      results <- tidyr::unnest(results_nest, op) # nolint
+      # `op` is necessary, so that other list columns are not unnested
+      # but lintr complains about `op` not being defined
```

Comment:
ah interesting and tidyr has no such thing like dplyr's `.data$`?

Reply:
Nope, or rather `op` is part of `...`?! I don't remember though if lintr would be satisfied with `op` being defined in `globalVariables`.